service/telemetry.go

```diff
@@ -84,22 +83,28 @@ func (tel *telemetryInitializer) init(buildInfo component.BuildInfo, logger *zap
 	var err error
 	tel.doInitOnce.Do(
 		func() {
-			err = tel.initOnce(buildInfo, logger, cfg, asyncErrorChannel)
+			if cfg.Metrics.Level == configtelemetry.LevelNone || cfg.Metrics.Address == "" {
+				logger.Info(
+					"Skipping telemetry setup.",
+					zap.String(zapKeyTelemetryAddress, cfg.Metrics.Address),
+					zap.String(zapKeyTelemetryLevel, cfg.Metrics.Level.String()),
+				)
+				return
+			}
+
+			err = tel.initOnce(buildInfo, logger, cfg)
+
+			go func() {
+				if serveErr := tel.server.ListenAndServe(); serveErr != nil && !errors.Is(serveErr, http.ErrServerClosed) {
+					asyncErrorChannel <- serveErr
+				}
+			}()
 		},
 	)
 	return err
 }
 
-func (tel *telemetryInitializer) initOnce(buildInfo component.BuildInfo, logger *zap.Logger, cfg telemetry.Config, asyncErrorChannel chan error) error {
-	if cfg.Metrics.Level == configtelemetry.LevelNone || cfg.Metrics.Address == "" {
-		logger.Info(
-			"Skipping telemetry setup.",
-			zap.String(zapKeyTelemetryAddress, cfg.Metrics.Address),
-			zap.String(zapKeyTelemetryLevel, cfg.Metrics.Level.String()),
-		)
-		return nil
-	}
-
+func (tel *telemetryInitializer) initOnce(buildInfo component.BuildInfo, logger *zap.Logger, cfg telemetry.Config) error {
```

Comment:
I've moved some of the code from `initOnce` in order to avoid starting a http server for the prometheus handler, since that is already captured by the tests in `collector_test.go`